lib/node_modules/@stdlib/math/base/special/expm1/src/main.c

```diff
@@ -0,0 +1,215 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* ## Notice
+*
+* The following copyright, license, and long comment were part of the original implementation available as part of [FDLIBM]{@link http://www.netlib.org/fdlibm/s_expm1.c}. The implementation follows the original, but has been modified for JavaScript.
+*
+* ```text
+* Copyright (C) 2004 by Sun Microsystems, Inc. All rights reserved.
+*
+* Developed at SunPro, a Sun Microsystems, Inc. business.
+* Permission to use, copy, modify, and distribute this
+* software is freely granted, provided that this notice
+* is preserved.
+* ```
+*/
+
+#include "stdlib/math/base/special/expm1.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/number/float64/base/get_high_word.h"
+#include "stdlib/number/float64/base/set_high_word.h"
+#include "stdlib/constants/float64/ninf.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib/constants/float64/exponent_bias.h"
+#include "stdlib/constants/float64/half_ln_two.h"
+#include <stdint.h>
+
+static const double OVERFLOW_THRESHOLD = 7.09782712893383973096e+02; // 0x40862E42 0xFEFA39EF
+
+// High and low words of ln(2):
+static const double LN2_HI = 6.93147180369123816490e-01; // 0x3FE62E42 0xFEE00000
+static const double LN2_LO = 1.90821492927058770002e-10; // 0x3DEA39EF 0x35793C76
+
+// 1 / ln(2):
+static const double LN2_INV = 1.44269504088896338700e+00; // 0x3FF71547 0x652B82FE
+
+// ln(2) * 56:
+static const double LN2x56 = 3.88162421113569373274e+01; // 0x4043687A 0x9F1AF2B1
+
+// ln(2) * 1.5:
+static const double LN2_HALFX3 = 1.03972077083991796413e+00; // 0x3FF0A2B2 0x3F3BAB73
+
+/* Begin auto-generated functions. The following functions are auto-generated. Do not edit directly. */
+
+// BEGIN: polyval_q
+
+/**
+* Evaluates a polynomial.
+*
+* ## Notes
+*
+* -   The implementation uses [Horner's rule][horners-method] for efficient computation.
+*
+* [horners-method]: https://en.wikipedia.org/wiki/Horner%27s_method
+*
+* @param x    value at which to evaluate the polynomial
+* @return     evaluated polynomial
+*/
+static double polyval_q( const double x ) {
+	return -0.03333333333333313 + (x * (0.0015873015872548146 + (x * (-0.0000793650757867488 + (x * (0.000004008217827329362 + (x * -2.0109921818362437e-7)))))));
+}
+
+// END: polyval_q
+
+/* End auto-generated functions. */
+
+/**
+* Computes `exp(x) - 1`.
+*
+* @param x    input value
+* @return	  output value
+*
+* @example
+* double v = stdlib_base_expm1( 0.0 );
+* // returns 0.0
+*/
+double stdlib_base_expm1( const double x ) {
+	double halfX;
+	int sign;
+	double lo;
+	uint32_t hi;
+	uint32_t hx;
+	double r1;
+	double xc = x;
```

Comment:
In the implementation, I don't see any assignment to `xc`, and, thus, I don't think `xc` is necessary in this function. All instances of `xc` can be replaced by `x`.